action.yaml

```
@@ -0,0 +1,68 @@
+---
+name: Shipwright Setup (CI)
+description: Installs Shipwright Build Controller, CLI and Container Registry
+branding:
+  color: blue
+  icon: anchor 
+inputs:
+  tekton-version:
+    description: Tekton Pipelines release version
+    required: false
+    default: v0.37.0
+  shipwright-ref:
+    description: Shipwright Build Controller repository reference, tag or SHA
+    required: false
+    default: v0.10.0
+  cli-ref:
+    description: Shipwright CLI repository reference, tag or SHA
+    required: false
+    default: v0.10.0
+  kind-cluster-name:
+    description: KinD cluster name, by default "kind"
+    required: false
+    default: kind
+  setup-registry:
+    description: When enabled, the action deploys a Container Registry instance
+    required: true
+    default: "true"
+runs:
+  using: composite
+  steps:
+    # pre-flight checks, making sure the dependencies needed for the upcoming steps are available
+    - shell: bash
+      run: ${{ github.action_path }}/probe.sh
+
+    # deploying the container registry when input flag is set, waiting for the deployment to reach
+    # ready status before proceeding
+    - shell: bash
+      if: ${{ inputs.setup-registry == 'true' }}
+      run: ${{ github.action_path }}/install-registry.sh
+
+    # deploying tekton pipline controller and dependencies, waiting for it to reach ready status
+    - shell: bash
+      env:
+        TEKTON_VERSION: ${{ inputs.tekton-version }}
+      run: ${{ github.action_path }}/install-tekton.sh
+
+    # checking out the build controller project locally to perform the rollout and inspection of the
+    # controller instance in the cluster
+    - uses: actions/checkout@v3
+      with:
+        repository: shipwright-io/build
+        ref: ${{ inputs.shipwright-ref }}
+        path: src/build
+      if: ${{ github.event.pull_request.head.repo.full_name != github.repository }}
```

I am not getting behind this `if` clause. Can we safely use `github.event.pull_request` ? Would not it be null if the event is just a push to a branch? I am also not understanding your comparison against `github.repository`. Assuming somebody uses Shipwright in his project, neither the PR nor the repo will be shipwright build = it must be pulled.

What I would understand, not sure if that's possible syntax-wise:

```suggestion
      if: ${{ github.event?.pull_request?.head.repo.full_name != "shipwright-io/build" && github.repository != "shipwright-io/build" }}
```

---

The comparison here is to assert if the current project should be subject to checkout, and that must never happen when the current project is the actual project. So, I think the `if` clause stands correct.

---

Hm, but this `if` is on the step that checks out Shipwright Build and not the current project.

---

The current project does not need checkout, therefore the `if` clause compares the current repository is "the" Shipwright Build project, and therefore should not be subject to checkout.

---

Right, that explanation = "compares the current repository is "the" Shipwright Build project" imo makes sense. But, I do not see how the code is doing that. If this action would for example be used in the CLI and would run for a PR from me, then `github.event.pull_request.head.repo.full_name` would be `shipwright-io/cli` and `github.repository` would be `SaschaSchwarze0/cli`. The condition would evaluate to true because they are different, so it would clone. But, I don't understand how the coded relates to the intented clause.

---

As the initial implementation, this `if` clause would only work inside the `shipwright-io` organization. We will need more for the fork use-case, I think that's where our different perspective is coming from.

---

Should we start as is and create a issue to track this next work item?